### DISCUSSION TITLE
Get handle for a userControl

### DISCUSSION THREAD

**Discussion by chuckgmthompson** [unoplatform/uno#18663] [Q&A]
I am trying to bind the ui output of a process (freerdp parent-window parm) to the windowId for a userControl.
In winforms I would just get the Handle for the Form or Control I wanted to use as the surface.
Not sure how to do that in Uno though.
Any help would be a appreciated

**Comment by DevTKSS** [comment]
I am coming from Winforms and vb.net before recently started learning uno plattform development, so in my opinion the "Handle" type of IntPtr in vb.net as far as I know in uno application should be the xamlRoot if I understood that correct so far :)

E.g.:
In a WinForms you could tell a WindowsForms.Forms.MessageBox (vb.net language) this:
```
vb

Dim Answer as DialogResult = MessageBox.Show(MyParentWindowHandle,title:="Your Title here",text:="Hello World!",buttons:=MessageBoxButtons.YesNo,icon:=MessageBoxIcon.Information)
```
and it would be happy (okay, title, buttons and icon are optional, but you know what I mean by that)

So in WinForms it would expect the MyParentWindowHandle as Win32Window.

In Uno App using a MessageBox (or is it called MessageDialog? nevermind) is outdated and only present for compatiblity as far as I read below Dialogs docs, you would do the same as mentioned above with writing a async void/Task in the MainPage (ViewModel did not work for me last days, opened a issue already) thats doing this:

```
public async void ShowContentDialog()
{
      var dialog = New ContentDialog()
      {
            Title = "Your Title here",
            Content = "Hello World!",
            PrimaryButtonText = "Yes",
            SecondaryButtonText  = "No",
            DefaultButton = PrimaryButton,
            XamlRoot = this.XamlRoot
       };
      var result = await dialog.ShowAsync();
      if (result == ContentDialogResult.Primary)
      {
            //Do something on Yes Click
      }
      else if (result == ContentDialogResult.Secondary)
      {
            //Do something on No Click
      }
      else
      {
            // This would be triggered in case the user closes the ContentDialog without using your Options
       };
}
```
Normally I would also tell you to use a XamlRootProvider (no standard Class but should really be), but if you dont use mvvm I would not recommend because in my opinion you should not need it when you can call this.XamlRoot and "this" is some control that do have a xamlRoot. you can also choose some control thats known (write the Name in a c# markup with Name(out var SomeKnownName)) and that should also work.

Hope this answers your question?

**Reply by chuckgmthompson** [reply]
thanks, but the this.XamlRoot.Handle is not set (0x000).
Is it posible to get the hwnd (Handle) of the active window?  or even a new window for that matter?
any help would be appreciated.

**Reply by chuckgmthompson** [reply]
In WPF, I do it like this and it works as this is what I am looking for.
 IntPtr windowHandle = new WindowInteropHelper(Application.Current.MainWindow).Handle;
 var valueIamLookingFor = windowHandle.ToInt64();

**Reply by DevTKSS** [reply]
🤔 in app.xaml.cs is a window, so you could go and get this if you are explicit wanting something thats called window, but i think it is simply called xaml root in uno, but I cant tell.
@morning4coffe-dev since you did the rootprovider last days, could you tell about this with more uno experience than me newbe?
